### ISSUE TITLE
Shellcheck is not run in project root or relative to the file

### ISSUE THREAD

**Issue by mafredri** [AtomLinter/linter-shellcheck#31]
I'm running to an issue where I have to specify the absolute path to the shellcheck source-directive when sourcing a file.

Example (works when modifying *linter-shellcheck* to run shellcheck with `-x` parameter):

```shell
# shellcheck source=/path/to/project/source/config/settings.sh
. ${PKG_DIR}/config/settings.sh
```

Expected to work (but does not):

```shell
# shellcheck source=source/config/settings.sh
. ${PKG_DIR}/config/settings.sh
```

Or (assuming this is located in file: `/path/to/project/source/lib/stuff.sh`):

```shell
# shellcheck source=../config/settings.sh
. ${PKG_DIR}/config/settings.sh
```

**Comment by Arcanemagus** [comment]
After looking back over this, #40 actually only fixed _part_ of this issue by setting the working directory to the directory of the file being linted. Is that sufficient for what you were looking for @mafredri? Or does this need to be re-opened and the working directory set to the project root (as Atom sees it), falling back to the current behavior if it can't find one?

**Comment by mafredri** [comment]
@Arcanemagus thanks for checking in, I do believe that's sufficient and makes sense when modifying files outside a project root (i.e. specifying a relative path from the file being edited).

**Comment by Arcanemagus** [comment]
Thanks, leaving this closed then 😉.
